### ISSUE TITLE
Unused errors can be attributed to the wrong line

### ISSUE THREAD

**Issue by markwainwright** [markwainwright/atom-linter-coffee-variables#4]
If a file contains a `defined but never used` error, the plugin highlights the first use of that variable name in the file as the culprit, which isn't accurate in files that use the same variable name in multiple places.

Example:

``` coffeescript
1   window.doSomething()
2    .then (response) -> response.foo
3    .then (response) -> 'bar'
```

Expected result:
- 'Unused' error on line 3

Actual result:
- 'Unused' error on line 2

**Comment by MocanuMarius** [comment]
if error.ruleId is 'no-unused-vars' or not positionFromSourcemap.line
    positionFromVariableLookup = _lookUpVariablePosition(variableName, variables)[0]

I guess these 2 lines ( 70 and 71 ) are the culprit for that behavior.

**Comment by markwainwright** [comment]
@MocanuMarius 
Yep, for sure. The problem is that coffeescript sourcemaps don't tell us where variables were defined – I think. So as a fallback, the package looks at all the variable tokens that coffeescript identified and returns the first matching one. 

Later this week, when I have time, I'll verify that the coffeescript sourcemap really doesn't have the necessary information and then work on whether we can discriminate between the variable tokens to highlight the right one.

**Comment by MocanuMarius** [comment]
![image](https://cloud.githubusercontent.com/assets/6069369/10933498/871a5c16-82e1-11e5-9646-b018ef39e234.png)

What about this one ? This is the whole file you see in the picture, for some reason the fact that it's an array triggers the unused variable thingie :/

**Comment by markwainwright** [comment]
That would be a different issue. And actually it doesn't have anything to do with the function being in an array.

This has the same warning:

``` coffeescript
module.exports = (src, dest) ->
  dest = src
```

And if you compile that to JS and run it through ESLint you'll see the same warning. I guess it's because you're changing the value of `dest` but still not _using_ it in the function body – instead, the function just returns `src` (remember the last statement of a function is always returned in CoffeeScript). A more usual map operation might be to return the new value instead of mutating the old one.

**Comment by markwainwright** [comment]
I just released v0.4.2 which should fix this, so I'll close this issue. If you still have problems, please feel free to re-open it. Thanks!
